src/shared/unit-file.c

```diff
@@ -21,3 +22,54 @@ bool unit_type_may_template(UnitType type) {
                       UNIT_TIMER,
                       UNIT_PATH);
 }
+
+int unit_validate_symlink(const char *filename, const char *target) {
+        const char *src, *dst;
+        UnitType t, t2;
+        int r, r2;
+
+        src = basename(filename);
+        dst = basename(target);
+
+        /* Reject cases where the alias or the target unit have instance names. */
+        r = unit_name_to_instance(src, NULL);
+        if (r < 0)
+                return log_notice_errno(r, "%s: not a valid unit name \"%s\": %m", filename, src);
+        if (r == UNIT_NAME_INSTANCE)
+                return log_notice_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "%s: unit symlink has instance name, rejecting.", filename);
```

Comment:
hmm, waaa? not sure i grok this? we have symlinks like "gettys.target/serial-getty@ttyS5.service" → "../serial-getty@.service" all the time, so why disallow this?

Reply:
This is only about *alias* symliks, not about symlinks in .wants or such. I'll rename the function to make this clearer.

The general idea was that symlinks like `/etc/systemd/system/serial-getty@ttyS1.service → anything` (apart from masking), are either pointless or invalid, so we should forbid them in general.

Reply:
hmm, should it be forbidden though?

I mean i see nothign wrong with having many aliases as long as the instance matches...

i.e. foobar@waldo.service and quux@waldo.service should be fine as aliases

Reply:
Do you think it would be OK to have foobar@waldo.service alias quux@waldo.service, and foobar@zoom.service alias foobar-alias@zoom.service? Or should foobar@waldo.service→quux@waldo.service automatically apply to all instances and foobar@.service should alias quux@.service ?